### PR TITLE
fix

### PR DESCRIPTION
**Pull request by michalrudolf**

### CODE REVIEW

File: src/wui/http/http_client.c [hunk 1]
```diff
@@ -533,11 +534,20 @@ wui_err buddy_http_client_init(uint8_t id, void * container) {
...
 
     req->recv_fn = data_received_fun;
-    tcp_connect(req->pcb, &host_ip4, 9000, httpc_tcp_connected);
+    tcp_connect(req->pcb, &host_ip4, CONNECT_DEF_PORT, httpc_tcp_connected);
     return ERR_OK;
 }
 
 void buddy_http_client_loop() {
+    static bool connect_ip_supplied = false;
+
```
Comment (joshymjose) [inline src/wui/http/http_client.c:570]: what happen when user once enabled, then disabled connect?

Comment (joshymjose) [reply src/wui/http/http_client.c:570]: need a work flow  how is  connect disbaled.

File: src/wui/http/http_client.c [hunk 2]
```diff
@@ -448,7 +449,7 @@ static const char * telemetry_to_http_str(const char * host_ip4_str){
 
     eeprom_get_string(EEVAR_CONNECT_KEY_START, printer_token, API_TOKEN_LEN);
     printer_token[API_TOKEN_LEN] = 0;
-    snprintf(header, HEADER_MAX_SIZE, "POST %s HTTP/1.0\r\nHost: %s\nPrinter-Token: %s\r\n", uri, host_ip4_str, printer_token);
+    snprintf(header, HEADER_MAX_SIZE, "POST %s HTTP/1.0\r\nHost: %s\nPrinter-Token: %s\r\n\r\n", uri, host_ip4_str, printer_token);
```
Comment (joshymjose) [inline src/wui/http/http_client.c]: we have to combine the header generation used in HTTP server replies

**Review by joshymjose — CHANGES_REQUESTED** [review]

File: src/wui/http/http_client.c (hunk 2 above)
Comment (michalrudolf) [reply src/wui/http/http_client.c]: I have already done that but in different branch: 
https://github.com/michalrudolf/Prusa-Firmware-Buddy/tree/A3IDES-687_STATE_CHANGED

File: utils/test_http_server/connect.py [hunk 3]
```diff
@@ -4,7 +4,24 @@
 from ipaddress import ip_address
 
 #HOST = '192.168.1.111'
-PORT = 9000
+PORT = 8000
+HTTP_OK = "HTTP/1.1 200 OK\r\n\r\n"
+HTTP_LOWLVL_GCODE = "HTTP/1.1 200 OK\r\nContent-Type: text/x.gcode\r\nContent-Length: 6\r\nCommand-Id: 1254\r\n\r\nG28 XY"
+HTTP_RESPONSE_ACC = "POST /p/events HTTP/1.0\r\nPrinter-Token: 01234567899876543210\r\nContent-Type: application/json\r\n\r\n{\"event\":\"ACCEPTED\",\"command_id\":1254}"
```
Comment (joshymjose) [inline utils/test_http_server/connect.py:10]: need to check and also handle situation when content-length is wrong (both number larger and smaller than actual)

**Review by joshymjose — APPROVED** [review]

**Pull request merged**